CMakeLists.txt

```diff
@@ -253,6 +251,16 @@ add_compile_options(/MP) # Build with Multiple Processes (number of processes eq
 add_compile_options(/GS) # Buffer Security Check
 add_compile_options(/Zm200) # Specify Precompiled Header Memory Allocation Limit of 150MB
 add_compile_options(/wd4960 /wd4961 /wd4603 /wd4627 /wd4838 /wd4456 /wd4457 /wd4458 /wd4459 /wd4091 /we4640)
+add_compile_options(/Zi) # enable debugging information
+add_compile_options(/O1) # minimize space
+
+if (CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL Relwithdebinfo)
+  add_compile_options(/GL) # enable WholeProgramOptimization
```

Comment:
Yes you right regarding the redundancy of /Zi and /O1 (the two lines just before the if statement). I actually realized that just after I PR’ed. However, regarding /O1 vs /O2. You said it results in better overall performance. If you’ve done any perf testing regarding this, the results are invalid now because WPO can make a significant difference. After enabling WPO, it might be better to use /O2.